zero_bin/rpc/src/main.rs

```diff
@@ -53,6 +53,14 @@ impl Cli {
             } => {
                 let checkpoint_block_number =
                     checkpoint_block_number.unwrap_or((start_block - 1).into());
+                // Ensure checkpoint_block_number is less than start_block
+                if let BlockId::Number(checkpoint_block_number) = checkpoint_block_number {
+                    if checkpoint_block_number.as_number().unwrap() >= start_block {
+                        anyhow::bail!(
+                            "Checkpoint block number must be less than the start block number"
```

Comment:
The error message could carry more information, ideally something like:
```suggestion
                            "Found checkpoint block number {} whereas range start is {}",
                            checkpoint_block_number,
                            start_block
```

Reply:
(and tweaked accordingly for the case where we have a previous proof)